app/models/concerns/spotlight/solr_document/active_model_concern.rb

```diff
@@ -10,7 +10,7 @@ module Spotlight::SolrDocument::ActiveModelConcern
   module ClassMethods
 
     def primary_key
-      :id
+      self.unique_key
```

Comment:
related note: there seems to be a similar problem in atomic_updates.rb. When I put the following in hash_for_solr_update, the solr update works. Otherwise, the solr update fails because it's trying to set: pid.
Hash[doc.map do |k,v|
        val = if k.to_sym == :pid